src/IROperator.cpp

```diff
@@ -1368,17 +1368,21 @@ Tuple tuple_select(const Expr &condition, const Tuple &true_value, const Tuple &
 }
 
 Expr mux(const Expr &id, const std::vector<Expr> &values) {
-    user_assert(values.size() >= 2) << "mux() only accepts values with size >= 2.\n";
+    user_assert(!values.empty()) << "mux() requires a non-empty vector of values";
+    if (values.size() == 1) {
+        // Useful in generic code where the size of the values vector
+        // might be degenerate.
+        return values[0];
```

Comment:
What does mux() do if the `id` Expression is out of range? (ie is there an assert-fail, or does it just always return the last/first/etc entry)? I presume it's analogous to multi-select and returns the final value for no-match, but I don't see this documented explicitly.

Reply:
Same as before this PR. I updated the comment in IROperator.h to be clear.